### DISCUSSION TITLE
Help needed, Can't generate AAB, getting error with AIR 50.2.5.1 :/

### DISCUSSION THREAD

**Discussion by AlikeStudio** [airsdk/Adobe-Runtime-Support#3259] [Q&A] [answered]
Hello comunity and contributors,
We had issues with Package Delivery ANE recently. We decided to update all the ANEs and AIR version to the latest (50.2.5.1) in order to solve those.
Now we can't generate AAB package with this version X(

We use this to compile the package, and we are getting always an error (log):
*** ADT Logging at 2024/05/20 20:52:00 ***
Called by: D:\AIR\AIRComp50.2\lib\adt.jar
ADT 50.2.5.1 called with: -package -target aab -storetype PKCS12 -keystore cert.p12 -storepass xxxxxx MyApp.aab  MyApp.xml -platformsdk C:\\Users\\User\\AppData\\Local\\Android\\Sdk -extdir libsAndroid -C MyApp.swf assetpack -C (...)
Creating device SDK handler - location passed in = C:\Users\User\AppData\Local\Android\Sdk
Version in APK = 50.2.5.1, version of ADT = 50.2.5.1
Version in APK = 50.2.5.1, version of ADT = 50.2.5.1
Version in APK = 50.2.5.1, version of ADT = 50.2.5.1
Version in APK = 50.2.5.1, version of ADT = 50.2.5.1
Gradle build where ADT is using JRE version 11.0.20
but where Gradle will be using the JDK from C:/Program Files/Java/jdk-11\bin\java.exe
Gradle build command:  C:/Program Files/Java/jdk-11\bin\java.exe -Xmx1024m -Dorg.gradle.appname=gradlew -Dorg.gradle.jvmargs=-Xmx1024m -Duser.dir="C:\Users\User\AppData\Local\Temp\296c9001-21a8-4b85-8ed4-ac5f99ebfa3f" -cp "C:\Users\User\AppData\Local\Temp\296c9001-21a8-4b85-8ed4-ac5f99ebfa3f\gradle\wrapper\gradle-wrapper.jar";"D:\AIR\AIRComp50.2\lib\android\bin\gradle-cli.jar" org.gradle.wrapper.GradleWrapperMain :app:bundleRelease
..java.base/java.lang.Thread.getStackTrace(Thread.java:1606)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.apk.AABOutputStream.buildGradle(AABOutputStream.java)
..com.adobe.air.apk.AABPackager.renameOutput(AABPackager.java)
..com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
..com.adobe.air.apk.AABPackager.createPackage(AABPackager.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
Unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed: ERROR:D8: com.android.tools.r8.kotlin.H
ERROR:D8: com.android.tools.r8.kotlin.H

FAILURE: Build failed with an exception.

Before the AIR version update we generated ok the AAB at least, but not now :_( (example of a previous compilation taken from adt.log as well):
*** ADT Logging at 2024/05/13 21:32:39 ***
Called by: D:\AIR\AIRComp50.2\lib\adt.jar
ADT 50.2.3.6 called with: -package -target aab -storetype PKCS12 -keystore cert.p12 -storepass xxxxxx MyApp.aab  MyApp.xml -platformsdk C:\\Users\\User\\AppData\\Local\\Android\\Sdk -extdir libsAndroid -C MyApp.swf assetpack -C (...)
Creating device SDK handler - location passed in = C:\Users\Edu\AppData\Local\Android\Sdk
Version in APK = 50.2.3.5, version of ADT = 50.2.3.6
Version in APK = 50.2.3.5, version of ADT = 50.2.3.6
Version in APK = 50.2.3.5, version of ADT = 50.2.3.6
Version in APK = 50.2.3.5, version of ADT = 50.2.3.6
Gradle build where ADT is using JRE version 11.0.20
but where Gradle will be using the JDK from C:/Program Files/Java/jdk-11\bin\java.exe
Gradle build command:  C:/Program Files/Java/jdk-11\bin\java.exe -Xmx1024m -Dorg.gradle.appname=gradlew -Dorg.gradle.jvmargs=-Xmx1024m -Duser.dir="C:\Users\Edu\AppData\Local\Temp\d0c82339-f099-4544-9d54-6ae1c711ee06" -cp "C:\Users\Edu\AppData\Local\Temp\d0c82339-f099-4544-9d54-6ae1c711ee06\gradle\wrapper\gradle-wrapper.jar";"D:\AIR\AIRComp50.2\lib\android\bin\gradle-cli.jar" org.gradle.wrapper.GradleWrapperMain :app:bundleRelease
AAB signer command:  C:/Program Files/Java/jdk-11\bin\jarsigner.exe -storetype PKCS12 -keystore cet.p12 -storepass xxxx C:\Users\User\AppData\Local\Temp\d0c82339-f099-4544-9d54-6ae1c711ee06/app/build/outputs/bundle/release/app-release.aab 1

Any help would be really (REALLY) appreciated,

Thanks for the attention

**Comment by marchbold** [comment]
Check out this discussion: https://github.com/airsdk/Adobe-Runtime-Support/issues/3236

Short answer, either: 
- set your minimum version to 24. 
- use the Android Studio Project output to build.
- wait for an update from Harman.

**Reply by ajwfrost** [reply]
Another option is to drop back the Android SDK and build tools to version 33...

If you do switch to using Android Studio, you can first update your gradle settings by adding the below into your `<android>` block in the app descriptor XML:
```
<gradleVersion>8.0</gradleVersion>
<androidGradlePluginVersion>8.1.1</androidGradlePluginVersion>
```
Then once the Android Studio project is created, you would need to go into the build.gradle file for any extension libraries, and add a `namespace` entry (e.g. if using the Play Asset Delivery ANE, you should see an extension library `com.google.android.play.corecommon`, so within the build.gradle for that project, inside the `android {` section, add
```
    namespace 'com.google.android.play.corecommon'
```

The changes we're making to ADT will
- add in the namespaces automatically (we'd already done it for the main 'app' but had missed the ANE libraries...)
- detect when you're using build tools or platform higher than 33, and update the Gradle/AGP required versions automatically

thanks

**Comment by AlikeStudio** [comment]
Thank you very much for your reply @marchbold and @ajwfrost  !!

We'll try with these suggestions.
We're not used to generate the packages using Android Studio. In the past we used to generate APK directly from IntelliJ. Now we are using a the command line to do this calling ADT directly.
Whatever you could do to do our life easier will be more than welcome :) (since we were AS3 devs in the past specialists in visual stuff, we're a little out of our confort zone with all that java/kotlin stuff... :_) )

Thanks again!

**Comment by AlikeStudio** [comment, marked as answer]
Hello again @marchbold and @ajwfrost,

Thank to your help now we can generate AAB packages again from command line (batch file).
With a couple of changes we have it working again:
1. Adding in `<android>` block:
```
<gradleVersion>7.5</gradleVersion>
<androidGradlePluginVersion>7.4.0</androidGradlePluginVersion>
```
2. And changing minSdkVersion from "21" to "24"

this last change seems we have lost 17% of devices compatibility (messsage in Google Play Console said) at the moment. We would recover some of them once you finish the updates your working on, @ajwfrost? We hope so :)
Anyway, we can now continue finding out what happen to us with Delivery Package methods at least
Still not getting expected results from "getAssetPackLocation" or "getAssetAbsolutePath", always getting null and status 0 (unknown) from the package, but that's another story...

Thanks A LOT for your support with this problem. That had us stucked!

**Reply by ajwfrost** [reply]
Hi

Yes, you'd be able to revert your `minSdkVersion` change once we've made our changes. But you could probably also get away with just keeping that lower, and uninstalling the Android-34 or Android-35 platform so that we use Android-33....

thanks
